src/genie/libs/parser/iosxe/show_snmp.py

```diff
@@ -79,3 +79,118 @@ def cli(self, output=None):
 
         return ret_dict
 
+
+
+# ===================
+# Schema for:
+#  * 'show snmp user'
+# ===================
+class ShowSnmpUserSchema(MetaParser):
+    """Schema for show snmp user."""
+
+    schema = {
+        Any(): {
+            Optional("access-list"): str,
+            "auth-protocol": str,
+            "engine-id": str,
+            "group-name": str,
+            "priv-protocol": str,
+            "storage-type": str
+        }
+    }
+
+
+# ===================
+# Parser for:
+#  * 'show snmp user'
+# ===================
+class ShowSnmpUser(ShowSnmpUserSchema):
+    """Parser for show snmp user"""
+
+    cli_command = "show snmp user"
+
+    def cli(self, output=None):
+        if output is None:
+            output = self.device.execute(self.cli_command)
+
+        else: 
+            output = output
+
+        # User name: SNMPv3-alfa
+        # Engine ID: 800000090300002790FBCF00
+        # storage-type: nonvolatile	 active
+        # Authentication Protocol: SHA
+        # Privacy Protocol: AES128
+        # Group-name: ALFA
+        #
+        # User name: nmsops
+        # Engine ID: 00000063000100A20A101B3E
+        # storage-type: nonvolatile	 active	access-list: 69
+        # Authentication Protocol: MD5
+        # Privacy Protocol: DES
+        # Group-name: nmcigroup
+
+        # User name: SNMPv3-alfa
+        user_capture = "(?P<user>\S+)"
+        p_user = re.compile("User\s+name:\s+{user_capture}".format(user_capture=user_capture))
+
+        # Engine ID: 800000090300002790FBCF00
+        engine_capture = "(?P<engine>\S+)"
+        p_engine = re.compile("Engine\s+ID:\s+{engine_capture}".format(engine_capture=engine_capture))
+
+        # storage-type: nonvolatile   active access-list: 69
+        storage_capture = "(?P<storage>\S+)"
+        access_capture = "(?P<access>.*)"
+        p_storage_access = re.compile("storage-type:\s{storage_capture}\s+active\s+access-list:\s+{access_capture}".format(storage_capture=storage_capture, access_capture=access_capture))
+
+        # storage-type: nonvolatile         active
+        p_storage = re.compile("storage-type:\s{storage_capture}\s+active".format(storage_capture=storage_capture))
+
+        # Authentication Protocol: SHA
+        auth_capture = "(?P<auth>\S+)"
+        p_auth = re.compile("Authentication\s+Protocol:\s+{auth_capture}".format(auth_capture=auth_capture))
+
+        # Privacy Protocol: AES256
+        priv_capture = "(?P<priv>\S+)"
+        p_priv = re.compile("Privacy\s+Protocol:\s+{priv_capture}".format(priv_capture=priv_capture))
+
+        # Group-name: group1
+        group_capture = "(?P<group>\S+)"
+        p_group = re.compile("Group-name:\s+{group_capture}".format(group_capture=group_capture))
+
+
+        snmp_user_obj = {}
+        current_loop_user = ""
+
+        for line in output.splitlines():
+            if p_user.match(line):
```

Comment:
Please add matching line as comment for each regex.

Reply:
Matching lines added.